examples/asr/jasper_infer.py

```diff
@@ -131,28 +163,95 @@ def main():
-                logprob.append(
-                    batch[j][:evaluated_tensors[4][i][j], :].cpu().numpy())
-        with open(args.save_logprob, 'wb') as f:
-            pickle.dump(logprob, f, protocol=pickle.HIGHEST_PROTOCOL)
+    if args.mode == 'infer':
+        if args.lm_path:
+            beam_hypotheses = []
+            # Over mini-batch
+            for i in evaluated_tensors[-1]:
+                # Over samples
+                for j in i:
+                    beam_hypotheses.append(j[0][1])
+
+            wer = word_error_rate(
+                hypotheses=beam_hypotheses, references=references)
+            logger.info("Beam WER {:.2f}".format(wer*100))
+
+        if args.save_logprob:
+            # Convert logits to list of numpy arrays
+            logprob = []
+            for i, batch in enumerate(evaluated_tensors[0]):
+                for j in range(batch.shape[0]):
+                    logprob.append(
+                        batch[j][:evaluated_tensors[4][i][j], :].cpu().numpy())
+            with open(args.save_logprob, 'wb') as f:
+                pickle.dump(logprob, f, protocol=pickle.HIGHEST_PROTOCOL)
+
+    if args.mode == 'eval':
+        if args.alpha_max is None:
+            args.alpha_max = args.alpha
+        # include alpha_max in tuning range
+        args.alpha_max += args.alpha_step/10.0
+
+        if args.beta_max is None:
+            args.beta_max = args.beta
+        # include beta_max in tuning range
+        args.beta_max += args.beta_step/10.0
+
+        beam_wers = []
+        checkpoints_loaded = False
+
+        for alpha in np.arange(args.alpha, args.alpha_max, args.alpha_step):
+            for beta in np.arange(args.beta, args.beta_max, args.beta_step):
+                logger.info(f'infering with (alpha, beta): ({alpha}, {beta})')
+                eval_tensors = [log_probs_e1, predictions_e1,
+                                transcript_e1, transcript_len_e1,
+                                encoded_len_e1]
+                beam_search_with_lm = nemo_asr.BeamSearchDecoderWithLM(
+                    vocab=vocab,
+                    beam_width=args.beam_width,
+                    alpha=args.alpha,
+                    beta=args.beta,
+                    lm_path=args.lm_path,
+                    num_cpus=max(os.cpu_count(), 1))
+                beam_predictions_e1 = beam_search_with_lm(
+                    log_probs=log_probs_e1, log_probs_length=encoded_len_e1)
+                eval_tensors.append(beam_predictions_e1)
+                if checkpoints_loaded:
+                    checkpoint_dir = None
+                else:
+                    checkpoint_dir = load_dir
+                    checkpoints_loaded = True
+
+                evaluated_tensors = neural_factory.infer(
+                    tensors=eval_tensors,
+                    checkpoint_dir=checkpoint_dir,
+                )
+
+                greedy_hypotheses = post_process_predictions(
```

Comment:
I think there is no need to add greedy_hypotheses here for each round evaluation of alpha & beta.

Reply:
@Slyne thanks for pointing it out. Removed it.